interceptors/logging/logging.go

```diff
@@ -37,34 +35,89 @@ var (
 	MethodTypeFieldKey   = "grpc.method_type"
 )
 
-func commonFields(kind string, typ interceptors.GRPCType, service string, method string) Fields {
+type fieldsCtxMarker struct{}
+
+var (
+	// fieldsCtxMarkerKey is the Context value marker that is used by logging middleware to read and write logging fields into context.
+	fieldsCtxMarkerKey = &fieldsCtxMarker{}
+)
+
+// Fields represents logging fields. It has to have even number of elements (pairs).
+type Fields []string
+
+func newCommonFields(kind string, c interceptors.CallMeta) Fields {
 	return Fields{
 		SystemTag[0], SystemTag[1],
 		ComponentFieldKey, kind,
-		ServiceFieldKey, service,
-		MethodFieldKey, method,
-		MethodTypeFieldKey, string(typ),
+		ServiceFieldKey, c.Service,
+		MethodFieldKey, c.Method,
+		MethodTypeFieldKey, string(c.Typ),
 	}
 }
 
-// Fields represents logging fields. It has to have even number of elements (pairs).
-type Fields []string
+// Iter returns FieldsIterator.
+func (f Fields) Iter() FieldsIterator {
+	// We start from -2 as we iterate over two items per iteration and first iteration will advance iterator to 0.
```

Comment:
Should we raise an error in case the length of `Fields` array is not even? We always expect the key-value format, so we can raise an error. Thoughts?